README.md

```diff
@@ -500,6 +500,35 @@ std::variant<fine::Ok<int64_t>, fine::Error<std::string>> find_meaning(ErlNifEnv
 Note that if you use a particular union frequently, it may be convenient
 to define a type alias with `using`/`typedef` to keep signatures brief.
 
+## Allocators
+
+For compatibility with the STL, fine supports stateless allocators when
+decoding values, while also supporting stateful allocators when encoding
+values. The following shows how a custom `MyAllocator` allocator and
+`my_memory_resource` memory resource can be used in conjunction with fine:
+
+```c++
+template<typename T>
+struct MyAllocator { ... };
+
+std::pmr::memory_resource* my_memory_resource = ...;
+
+std::vector<std::pmr::string, MyAllocator<std::pmr::string>> duplicate(
```

Comment:
Should the return also be `std::pmr::vector`?

I would also be fine not having this section, because it might be one of the things where people who would think about using allocators would already know about those things. In this PR we just generalise the templates, so it's closer to how they would use the STL anyway?

Reply:
No, I wanted a mix of stateless (`std::vector`) and stateful allocations (`std::pmr::string`) to ensure that both can be encoded correctly.

Reply:
> [...] I wouldn't show code that the user wouldn't actually ever write.

Also links to https://github.com/elixir-nx/fine/pull/5#discussion_r2139917786.

I will make the change.

Reply:
Yes! What I mean is that the example still shows a mix of both allocators (`std::vector` and `std::pmr::string`), which is probably not something they would do in practice?

In other words, I would mention stateless allocators and make the example use `std::vector` and `std::string`, both with `MyAllocator`. Then, below, I would have the note about about `polymorphic_allocator` and `get_default_resource`.

Does that make sense?